### PR TITLE
App display board

### PR DESCRIPTION
**Pull request by paulSanchezMarmelab**
IMG_20230503_100855

- [X] Display board tiles
- [X] Make board zoomable
- [X] Display icons
- [X] Display last played tile indicator
- [X] Display played tiles

### CODE REVIEW

**Comment by fzaninotto** [comment]
Needs rebase

File: mobileapp/assets/book.svg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
+<svg width="800px" height="800px" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (adguernier) [inline mobileapp/assets/book.svg:3]: This comment is useless

File: mobileapp/components/board/BoardRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import { StyleSheet, View } from "react-native";
+import { Board, GameState } from "@kamon/core";
+import TileComponent from "../tile/TileRenderer";
+import { ReactNativeZoomableView } from "@openspacelabs/react-native-zoomable-view";
+import { Circle, Line, Path, Rect, Svg } from "react-native-svg";
```
Comment (adguernier) [inline mobileapp/components/board/BoardRenderer.tsx:6]: These imports are useless

File: mobileapp/components/board/BoardRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    flexDirection: "row",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  nullableTile: {
+    width: "1",
+  },
+});
+
+type boardProps = {
```
Comment (adguernier) [inline mobileapp/components/board/BoardRenderer.tsx:28]:
```suggestion
type BoardProps = {
```

File: mobileapp/components/board/BoardRenderer.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  board: Board;
+  gameState: GameState;
+};
+
+const BoardRenderer = (props: boardProps) => {
```
Comment (adguernier) [inline mobileapp/components/board/BoardRenderer.tsx:33]:
```suggestion
const BoardRenderer = ({board, gameState}: BoardProps) => {
```

File: mobileapp/components/board/BoardRenderer.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  const { board = [], gameState = {} } = props;
+
+  const renderTilesFromLine = (line, x) =>
+    line.map((tile, y) =>
+      tile != null ? (
+        <TileComponent
```
Comment (adguernier) [inline mobileapp/components/board/BoardRenderer.tsx:39]:
```suggestion
        <Tile
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]
A few cosmetic changes, but we're close!

File: mobileapp/components/tile/TileRenderer.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  cyan: "#96d2d9",
+  green: "#14d990",
+  red: "#f24968",
+  blue: "#6929f2",
+  yellow: "#f2b807",
+  magenta: "#9b72f2",
+  grey: "grey",
+};
+
+const ICONS_SIZE = 20;
```
Comment (adguernier) [inline mobileapp/components/tile/TileRenderer.tsx:44]: Unused const here

File: mobileapp/components/tile/TileRenderer.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { StyleSheet, View } from "react-native";
+import { Tile, TileCoordinate } from "@kamon/core";
+import Hexagon from "../shapes/Hexagon";
+import { Svg, Circle, Path, G } from "react-native-svg";
```
Comment (adguernier) [inline mobileapp/components/tile/TileRenderer.tsx:5]: Useless import here 😬

File: mobileapp/components/tile/TileRenderer.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  A: <Ghost />,
+  B: <Dragon />,
+  C: <Dice />,
+  D: <Book />,
+  E: <Hat />,
+  F: <GamePad />,
+  O: <View></View>,
+};
+
+type tileProps = {
```
Comment (adguernier) [inline mobileapp/components/tile/TileRenderer.tsx:56]:
```suggestion
type TileProps = {
```

File: mobileapp/components/tile/TileRenderer.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  tile: Tile;
+  coordinates: TileCoordinate;
+};
+
+const TileRenderer = (props: tileProps) => {
```
Comment (adguernier) [inline mobileapp/components/tile/TileRenderer.tsx:61]:
```suggestion
const TileRenderer = ({ tile, coordinates }: TileProps) => {
```

File: mobileapp/assets/book.svg (hunk 1 above)
Comment (adguernier) [reply mobileapp/assets/book.svg:3]: I don't understand why we need SVG and components ? I don't see these assets used elsewhere in the app 🧐

File: mobileapp/components/shapes/Hat.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import { View } from "react-native";
+import Svg, { Path } from "react-native-svg";
+
+const Hat = () => {
+  return (
+    <View
```
Comment (adguernier) [inline mobileapp/components/shapes/Hat.tsx:6]: Can we found a better way to create these components ? There is a lot of repetition, maybe we can make a more generic component and pass props to it ?

Comment (paulSanchezMarmelab) [reply mobileapp/components/shapes/Hat.tsx:6]: This should be replaced by icons later on if we can when we stop having loading issues with them.

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**